lib/stimulus_reflex/logger.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module StimulusReflex
+  class Logger
+    attr_accessor :reflex, :current_operation
+
+    def initialize(reflex)
+      @reflex = reflex
+      @current_operation = 1
+    end
+
+    def print
+      return unless config_logging.lambda?
+      puts
+      reflex.broadcaster.operations.each do
+        puts config_logging.call(self) + "\e[0m"
+        @current_operation += 1
+      end
+      puts
+    end
+
+    private
+
+    def config_logging
+      return @config_logging if @config_logging
+
+      StimulusReflex.config.logging.binding.eval("using StimulusReflex::Utils::Colorize")
+      @config_logging = StimulusReflex.config.logging
```

Comment:
I liked `StimulusReflex.config.logging` being a method. Especially if you take into account the fact we call it twice.

Reply:
We call it twice in the same file, but don't mutate it or need it memoized... I dunno, seems like overkill, no?

Reply:
Well, we actually do memoize it as you explained [here](https://github.com/hopsoft/stimulus_reflex/pull/337/files#r523287037), don't we? Personally, I find this method harder to read.

Reply:
We need the lambda to be  `using Colorize`, so we need to do the `.binding.eval` somewhere. 

We could put it in the `configuration.rb` and have a special setter that does the `.binding.eval`. I shortly went that route, but then thought the context is clearer if do the binding stuff is where the logger calls it. It is more of a concern of the logger than the configuration, that this config should be colorizable.

I then memoized it, because not every call to the logger should again do the `binding.eval("using …")`, and it's probably not great for performance to do the `binding.eval` every time a reflex gets logged. We mutate its binding, so yes I think we need to memoize it or have a setter that mutates the binding!

I would be fine to move the binding stuff into the `configuration.rb`. Would have the advantage that I can change `config.logging` at runtime (though I would not know why you would want to do that, well actually you might want to play around on the console, while finding your perfect log formatting :-) )

Reply:
btw: By setter I mean a writer, something like this:
```ruby
class Configuration
  def logging= (lambda)
    @logging = lambda.binding.eval("using StimulusReflex::Utils::Colorize")
  end
end

Reply:
Sleeping on it, I think it would be more ergonomic to do it with the writter in the config. I myself found a bit annoying having to restart the server, when changing the logging lambda, to see the changes reflected.
@piotrwodz Do you want to do that?

Reply:
@RolandStuder Tho minds think alike 👍

Yes! It is definitely going to improve things. Not only the fact of changing lambda without a restart but also currently `config_logger` method is responsible for two things and I would like to change it. It should be memoizing `StimulusReflex.config.logging` lambda only.

I would like to do that! Thanks!

Reply:
@piotrwodz Memoizing will no longer be necessary in `Logger` if you adjust the writer. Just make sure, that in the `def initialize` you also do the binding, or better yet you also use the setter/writer. `@logging = -> {}` does not call `def logging=`

Reply:
@RolandStuder I spent some time on it ☝️

Unfortunately, this ain't going to work out. We cannot define a setter like proposed above. Why? Because it does not refer to our lambda anymore. It points to its binding instead which as a result causes `print` to crash.

What I decided to do then was separation of `StimulusReflex.config.logging` memoization and setting config logging binding.